src/ui-kit/form-controls/autocomplete/autocomplete.component.ts

```diff
@@ -134,6 +134,12 @@ export class SamAutocompleteComponent
   @Input() public freeTextSubtext: string = 'search';
 
   @Input() public isKeyValue?: boolean;
+
+  /**
+   * Limits number of items rendered immediately for faster performance.
+   * As user scrolls through list, more items are added if available.
+   */
+  @Input() public enableLazyRendering: boolean = false;
```

Comment:
New input option to handle large autocomplete suggestions - Render only `STARTING_MAX_ITEMS` initially, then gradually increment by `STARTING_MAX_ITEMS` whenever user scrolls at least halfway through the list.